### PR TITLE
mpris support for linux

### LINKED ISSUE #404: Use MPRIS for Linux
On Linux you can use an interface called MPRIS2 for controlling music its a freedesktop.org (http://www.freedesktop.org/wiki/Specifications/mpris-spec/metadata/) standard and used in all major desktop enviroments (Unity, Gnome, Plasma).
A NodeWebkit App which uses MPRIS2 is here https://github.com/twolfson/google-music-electron/ That would be nice for all linux users since it would also enable keyboard shortcuts, etc...

Greetings

### PR DESCRIPTION
**Pull request by jakejarrett**
custom build of dbus for nwjs 0.12.3

This allows Linux builds to have media key support & all the benefits of mpris integration (EG/ Have currently playing in panel)

@weblancaster Would it be best if I put a custom build command into grunt to exclude the mpris-service folder from windows/mac builds?

Closes #404

### CODE REVIEW

File: app/index.html [hunk 1]
```diff
@@ -233,6 +233,7 @@ <h4 id="playerUser" class="player_user" ng-click="goToUser($event)"></h4>
 <script src="public/js/common/SCapiService.js"></script>
 <script src="public/js/common/playerService.js"></script>
 <script src="public/js/common/notificationFactory.js"></script>
+<script src="public/js/mpris/mprisService.js"></script>
```
Comment (weblancaster) [inline app/index.html]: @jakejarrett can you move this module to common instead of mpris please.

File: app/public/js/mpris/mprisService.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+var gui = require('nw.gui');
+
+app.factory('mprisService', function($rootScope, $log, $timeout, $window, $state) {
+    if(process.platform !== "linux") {
+        return false;
+    } else {
```
Comment (weblancaster) [inline app/public/js/mpris/mprisService.js]: no need for else here.. 

you can do like this
```javascript
if ( process.platform !== "linux" ) {
   return false;
}

// then the code here

```

**Comment by Pitros** [comment]
There's no need to include `node_modules` folder with `mpris`, adding it to dependencies in `package.json` is enough.

**Comment by weblancaster** [comment]
You are right @Pitros can you make those changes please @jakejarrett

**Comment by jakejarrett** [comment]
For mpris to work, it needs to be rebuilt for nwjs (nw-gyp) I can add a grunt task, but it will require nw-gyp & [its dependencies](https://github.com/nwjs/nw-gyp#installation)

**Comment by weblancaster** [comment]
That's fine @jakejarrett just add in the package.json.. no need for the node_modules.

**Comment by weblancaster** [comment]
Hey @jakejarrett can you remove the node_modules from the commit please? thanks

**Comment by jakejarrett** [comment]
Removed node_modules from the commit.

**Comment by jakejarrett** [comment]
@weblancaster To reduce the amount of dependencies for building, I'll exclude windows from building dbus until nw-gyp doesn't depend on Visual Studio being installed. Is this alright?

**Comment by weblancaster** [comment]
@jakejarrett can you fix the conflicts please.. I will merge as soon as they get fixed. thanks

**Comment by jakejarrett** [comment]
@weblancaster I'll update the pull request tonight.

**Comment by weblancaster** [comment]
Thanks and sorry for the conflict

**Comment by jakejarrett** [comment]
@weblancaster Updated. Also, included the requirements in the commit message. It will throw an error during build if these are not met.

**Comment by weblancaster** [comment]
Great thanks @jakejarrett I'm going to re-test tonight.

**Comment by jakejarrett** [comment]
@weblancaster have you tested it yet? I found a bug while testing on another PC and wanted to check if i'd missed something

**Comment by jakejarrett** [comment]
I've updated the pull request as i'd overlooked dbus not being on os x.

**Comment by weblancaster** [comment]
@jakejarrett great.. I was going to merge but I noticed since this is just for Linux.. can we add a condition where the build would add this package just for that OS?

**Comment by jakejarrett** [comment]
@weblancaster Don't merge yet, with the new node-dbus update, It will crash due to this error

```
home/jake/dev/soundnode-app/dist/Soundnode-App/linux64/Soundnode-App: symbol lookup error: /tmp/.org.chromium.Chromium.qwqXka/node_modules/mpris-service/node_modules/dbus/build/Release/dbus.node: undefined symbol: _ZN6Signal14DispatchSignalEPN2v86HandleINS0_5ValueEEE
```

**Comment by weblancaster** [comment]
Oh ok.. thanks for letting me know @jakejarrett

**Comment by jakejarrett** [comment]
@weblancaster New node-dbus update seems to have resolved the symbol lookup error. Will update the pr

**Comment by jakejarrett** [comment]
Will revisit this when soundnode gets upgraded to newer nwjs or electon. node-dbus breaks on the current node version.

**Pull request closed without merging**